src/main.rs

```diff
@@ -41,15 +43,26 @@ fn pipeline_main() {
 }
 
 fn wait_for_threads(pipes: Vec<PipelineTracker>) {
-    for pipe in pipes {
+    println!("Waiting for threads to finish by sending end of work signal.");
+    for pipe in &pipes {
         let done_signal = FileWorkItem {
             file: String::new(),
             no_more_work: true,
         };
-        match pipe.transmit_channel.send(done_signal) {
-            Ok(_) => (),
-            Err(e) => println!("Couldn't send to channel: {}", e),
+        println!("Attempting to send stop signal to {:?}", pipe.thread);
+        let mut done = false;
+        while !done {
+            match pipe.transmit_channel.try_send(done_signal.clone()) {
```

Comment:
Somehow we can get a thread that stops processing but hasn't hung up, so this reports a full channel, not a thread that hung up.  But no thread is doing work.